cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -133,6 +133,7 @@
   private static final String NAMESPACE_CREATION_ENABLED = "master.environment.k8s.namespace.creation.enabled";
   private static final String CDAP_NAMESPACE_LABEL = "cdap.namespace";
   private static final String RESOURCE_QUOTA_NAME = "cdap-resource-quota";
+  private static final Set<String> BOOTSTRAP_CDAP_NAMESPACES = ImmutableSet.of("default", "system", "cdap");
```

Comment:
Is `cdap` a bootstrap namespace created by CDAP? If so, in what situations is it used?

Reply:
Not really sure what it's used for (and it's technically not a [bootstrap step](https://github.com/cdapio/cdap/blob/163ade54065ecac3a4fd80fe3d76a5aee38474f9/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/bootstrap/BootstrapStep.java#L75)), but it is [documented as a reserved namespace](https://cdap.atlassian.net/wiki/spaces/DOCS/pages/480313725/Namespaces#Namespace-Components) and is checked in [DefaultNamespacePathLocator](https://github.com/cdapio/cdap/blob/da58037c74af26ef26d832435b328782bbc3c443/cdap-common/src/main/java/io/cdap/cdap/common/namespace/DefaultNamespacePathLocator.java#L85-L94)/[NamespaceHttpHandler](https://github.com/cdapio/cdap/blob/da58037c74af26ef26d832435b328782bbc3c443/cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/NamespaceHttpHandler.java#L160-L164), so I figured it would be best to also treat it as a special case

Reply:
Understood, thanks!

Reply:
In fact, it is better to refactor the `NamespaceHttpHandler.isReserved` method into a static method in `NamespaceId`.

Reply:
Did the former suggestion to avoid having `cdap-kubernetes` depend on `cdap-app-fabric`

Reply:
nit: LGTM pending a minor refactor of `isReserved` to a static method within the `NamespaceId` class. Thanks!